spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkAction.java

```diff
@@ -163,7 +163,7 @@ protected Table newStaticTable(TableMetadata metadata, FileIO io) {
   }
 
   protected Dataset<Row> buildManifestFileDF(Table table) {
-    return loadMetadataTable(table, ALL_MANIFESTS).selectExpr("path as file_path");
+    return loadMetadataTable(table, ALL_MANIFESTS).selectExpr("path as file_path").distinct();
```

Comment:
I think this is worth a separate PR. @karuppayya, could you submit one?

Reply:
Also, could you dedup metadata locations in `buildOtherMetadataFileDF` in that PR too?

Reply:
I have created another PR: https://github.com/apache/iceberg/pull/2452

Reply:
Do we always want to distinct on this function?  I was just wondering because I think this won't collapse into a single join and basically would make the expire snapshots require 3 shuffles instead of just 1.

Reply:
Basically I think distinct should be called by the caller after making the union if that is required, rather than always calling distinct